packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -119,30 +106,8 @@ public View getView() {
     return mapView;
   }
 
-  void init(State lifecycleState) {
-    switch (lifecycleState) {
-      case RESUMED:
-        mapView.onCreate(null);
-        mapView.onStart();
-        mapView.onResume();
-        break;
-      case STARTED:
-        mapView.onCreate(null);
-        mapView.onStart();
-        break;
-      case CREATED:
-        mapView.onCreate(null);
-        break;
-      case DESTROYED:
-      case INITIALIZED:
-        // Nothing to do, the activity has been completely destroyed or not yet created.
-        break;
-    }
-    if (lifecycle != null) {
-      lifecycle.addObserver(this);
-    } else {
-      getApplication().registerActivityLifecycleCallbacks(this);
-    }
+  void init() {
+    lifecycleProvider.getLifecycle().addObserver(this);
```

Comment:
Not sure so double checking - will we get the initial notification for the current state?

Reply:
Yes, from `addObserver`'s documentation:

> The given observer will be brought to the current state of the LifecycleOwner. For example, if the LifecycleOwner is in `State#STARTED` state, the given observer will receive `Event#ON_CREATE`, `Event#ON_START` events